app/export.go

```diff
@@ -677,7 +685,15 @@ func (a *App) buildFavoritedByList(channelID string) ([]string, *model.AppError)
 func (a *App) exportAllDirectPosts(ctx request.CTX, writer io.Writer, withAttachments bool) ([]imports.AttachmentImportData, *model.AppError) {
 	var attachments []imports.AttachmentImportData
 	afterId := strings.Repeat("0", 26)
+	var postProcessCount uint64
+	logCheckpoint := time.Now()
+
 	for {
+		if time.Since(logCheckpoint) > 5*time.Minute {
```

Comment:
I think instead of a time based checkpoint, it might be better to have it based on number of posts exported. So we can keep the `postProcessCount`, but just change this check to something like `if postProcessCount % 10000 == 0`

Reply:
Yeah, I thought about this type of logging as well, but was trying to find a good balance between log noise and still logging if the export had slowed down to a crawl. The time-based approach seems to handle both situations well, but I will change it to this if you think it's better.

Reply:
I think since it's debug, so log noise shouldn't be an issue. This is more consistent with our other jobs like ES indexing and compliance export. But 1/5 on my side. I'll leave it to you.